ReactCommon/react/renderer/mapbuffer/MapBufferBuilder.cpp

```diff
@@ -122,9 +124,9 @@ MapBuffer MapBufferBuilder::build() {
   // Create buffer: [header] + [key, values] + [dynamic data]
   auto bucketSize = buckets_.size() * sizeof(MapBuffer::Bucket);
   auto headerSize = sizeof(MapBuffer::Header);
-  uint32_t bufferSize = headerSize + bucketSize + dynamicData_.size();
+  auto bufferSize = headerSize + bucketSize + dynamicData_.size();
 
-  header_.bufferSize = bufferSize;
+  header_.bufferSize = static_cast<uint32_t>(bufferSize);
```

Comment:
Hm, would it help to have `int32_t bufferSize` in header to avoid this conversion?

Reply:
Tinkered around with it. It looks like the culprit is that the variables are summed to make up bufferSize are of type size_t which size varies (32 or 64) depending on the machine architecture causing the type conversion warning. We could make header.bufferSize of type size_t to avoid the cast but then the header will change size based on the machine architecture which I don't think would be desired.